server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateCleanupManager.java

```diff
@@ -344,6 +360,14 @@ void deleteStalePaths(String clusterName, String clusterUUID, List<String> stale
         );
     }
 
+    void deleteStaleIndexRoutingPaths(List<String> stalePaths) throws IOException {
+        logger.debug(() -> String.format(Locale.ROOT, "Deleting stale index routing files from remote - %s", stalePaths));
+        getBlobStoreTransferService().deleteBlobs(
+            BlobPath.cleanPath(),
+            stalePaths
+        );
```

Comment:
BlobStore can be different for cluster state and routing table. Lets move this logic to RemoteRoutingTableService and use the blobStore from that.

Reply:
Makes sense, moved to RemoteRoutingTableService Class.